docs/adr/0008-use-tracing.md

```diff
@@ -0,0 +1,165 @@
+  There is a `tracing_log` crate which should mean we don't need to move all crates, but using this
+  crate does have some limitations, and given the move to `tracing` is largely mechanical we should probably
+  avoid it if we can - but it's a reasonable fallback if we need log output from crates we don't control.
+
+* All techniques we have for reporting log events and error reports to the application should be replaced with
+  a new mechanism based on [`tracing-subscriber`](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/)
+  to move logs and error-reports across the FFI.
+
+* This same mechanism, and the same raw data, should be used on all platforms (ie, Android, iOS and Desktop).
+
+## Implementation: Move to tracing for all app-services crates.
+
+This section describes the changes necessary to implement the above.
+
+### Move all our crates to `tracing`
+
+Most crates tend to use `log::debug!()` (or `trace!`, `error!` etc). We'd just change the crate to depend on `tracing` instead of `log`,
+and probably change each module to have as a prelude `use tracing::{trace, debug, info, error};` and change each `log::error!` to `error!`.
+Or maybe we just spell it out entirely as `tracing::log!(...)` - this is a choice for individual crate maintainers.
+
+Many tests start with `env_logger::try_init().unwrap()` or similar - these would all be replaced with the more-verbose but equivalent
+
+```
+        tracing_subscriber::registry()
+            .with(fmt::layer())
+            .with(EnvFilter::from_default_env())
+            .init();
+```
+
+(probably via a helper of some sort)
+
+### Implement a `tracing_subscriber::Layer`-based "subscriber" mechanism.
+
+This subscriber mechanism would require the application to know all tracing `targets` it cares about - and by default,
+each crate is its own target. The app must explicitly "subscribe" to all such targets individually.
+It will *not* be possible to subscribe to all events for performance reasons, and it's unlikely we'll allow
+any "matching" capabilities (eg, regular expressions or similar) - each target will use an exact string match.
+
+When subscribing to a target, the app will also need to specify the max-level for that target. In other words, a
+subscription can be considered a tuple of `(target, level)`.
+
+This implies that checking whether an individual tracing event has a subscriber will be a `HashMap` lookup for the target name,
+then a numeric comparison for the level. Each target will allow only one subscriber, identified by the `target`.
+This means that changing the level for an existing subscriber will involve replacing the subscription with a new one which holds the new level.
+It is expected that the implementation will simply be "set a new subscription", which will replace the old subscription.
+
+This assumes that the application will subscribe with the actual level needed by that subscriber - ie, that apps will not
```

Comment:
Are we likely to be the ones doing the plumbing on the application side (Desktop and mobile) to make sure the logs are going to the right place, too?

I think the answer is "yes", so we'll be the ones making sure that the application is subscribing correctly—but WDYT? 😄

Reply:
That's certainly true initially - although in the future we never know how this is going to work out - eg, I see no reason other teams will not start landing Rust code with their own requirements, so I'm mildly reluctant to state that as an absolute fact. Do you have a suggestion for a change here?

Reply:
Oh, no—it was a question, not a suggestion for edits 😅